docs/proposals/bearer-token.md

```diff
@@ -0,0 +1,123 @@
+# Bearer token authentication proposal
+
+## Introduction
+
+The service catalog currently supports basic auth as well as bearer token
+authentication. The broker will need to be enhanced to support bearer token
+authentication.
+
+## Problem Description
+The broker will need to support bearer token authentication.
+
+## <Implementation Details>
+Bearer token authentication uses the `Authorization` header followed by `Bearer`
++ 1 space + a base64 encoded token, see [RFC 6750 Section 2.1] [1]
+
+The broker will have a new `BearerAuth` struct that implements the [Provider] [2]
+interface.
+
+A new `TokenService` interface that will validate tokens either via Bearer
+token, shared key, or against a external service, etc. Basically used to deal with
+non username/password authentication.
+
+This new adapter will handle Bearer token and possibly SSL certificate
+authentication. The original UserService would handle BasicAuth, OAuth and any
+other model where a `UserPrincipal` will be returned.
+
+
+```golang
+type TokenService interface {
+   Validate(token string) bool
+}
+```
+
+There will be specific implementations of the TokenService which will validate
+the token. One simple one will simply read the token from the filesystem.
+Another one could connect to a service to validate the token.
+
+
+```golang
+// FileTokenService - reads a token from a static file
+type FileTokenService struct {
+   tokenFile string
+}
+
+func (f FileTokenService) Validate(token string) bool {
+    // read the file from tokenFile
+    // compare the token from the file to the one given
+    savedToken, err := ioutil.ReadFile(tokenFile)
+    if err != nil {
+        return false
+    }
+
+    return token == savedToken
+}
+```
+
+If we have an external service to handle validation something like this would be
+used:
+
+```golang
+type SomeExternalService struct {
+    service SomeAuthService
+}
+
+func NewSomeExternalService(service SomeAuthService) SomeExternalService {
+    return SomeExternalService{service: service}
+}
+
+func (s SomeExternalService) Validate(token string) bool {
+    return s.service.Validate(token)
+
+    // this could also be a simple get the value from the
+    // service, then compare it. Or much more complex if need be.
+    // Basically the idea is the AuthProviders calling Validate
+    // don't really care.
+}
+```
+
+Basic questions about the impact to Broker and APBs:
+
+ - How will the broker's behavior change?
+
+   The broker will now have 2 ways for authentication: basic auth and bearer
+   token. This can continue to expand as we see the need.
+
+ - Will this change APBs?
+
+   The bearer token will have no affect on the APBs.
+
```

Comment:
I'll update the proposal to answer your question, but both basic and bearer will be enabled in catasb but for `make run` all auth is currently disabled.